CVE export fails when Harbor role is assigned by OIDC group instead of user

We are running into an issue where users always get the following error when they try to generate a CVE export report for their project:
![image](https://user-images.githubusercontent.com/6350295/212946924-82a396d5-9f8c-4cf1-81d7-4890f4609a3d.png)

**Note** that the user has the right Harbor role for the specific project (Project Admin) and that the project contains repositories with images that contain CVEs .We don't use any filters.

The following is log data of the jobservice pod when we execute the CVE export through the GUI
```
2023-01-17 15:00:50	
{"log":"2023-01-17T14:00:49Z [ERROR] [/jobservice/runner/redis.go:123]: Job 'SCAN_DATA_EXPORT:0cb9aa60bae2323ef4bbca5d' exit with error: run error: http status code: 500, body: {\"errors\":[{\"code\":\"UNKNOWN\",\"message\":\"unknown error\",\"detail\":{\"RequestFailure\":{}}}]}\n","stream":"stdout","time":"2023-01-17T14:00:49.893878883Z"}
2023-01-17 15:00:49	
{"log":"2023-01-17T14:00:49Z [ERROR] [/jobservice/job/impl/scandataexport/scan_data_export.go:115]: Export Job Id = 23. Error when persisting report file /var/scandata_exports/scandata_export_23.csv to persistent storage: http status code: 500, body: {\"errors\":[{\"code\":\"UNKNOWN\",\"message\":\"unknown error\",\"detail\":{\"RequestFailure\":{}}}]}\n","stream":"stdout","time":"2023-01-17T14:00:49.882638538Z"}
2023-01-17 15:00:49	
{"log":"2023-01-17T14:00:49Z [ERROR] [/pkg/systemartifact/manager.go:111]: Error creating system artifact record for scan_data_export/scandata_export_23/sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855: http status code: 500, body: {\"errors\":[{\"code\":\"UNKNOWN\",\"message\":\"unknown error\",\"detail\":{\"RequestFailure\":{}}}]}\n","stream":"stdout","time":"2023-01-17T14:00:49.88185757Z"}
2023-01-17 15:00:49	
{"log":"2023-01-17T14:00:49Z [INFO] [/pkg/config/rest/rest.go:47]: get configuration from url: https://harbor-core:443/api/v2.0/internalconfig\n","stream":"stdout","time":"2023-01-17T14:00:49.257803727Z"}
2023-01-17 15:00:49	
{"log":"2023-01-17T14:00:49Z [INFO] [/jobservice/job/impl/scandataexport/scan_data_export.go:111]: Export Job Id = 23. CSV file size: 0\n","stream":"stdout","time":"2023-01-17T14:00:49.256448513Z"}
2023-01-17 15:00:49	
{"log":"2023-01-17T14:00:49Z [INFO] [/jobservice/job/impl/scandataexport/scan_data_export.go:105]: Creating repository for CSV file with blob : scandata_export_23\n","stream":"stdout","time":"2023-01-17T14:00:49.256388416Z"}
2023-01-17 15:00:49	
{"log":"2023-01-17T14:00:49Z [INFO] [/jobservice/job/impl/scandataexport/scan_data_export.go:95]: Export Job Id = 23, FileName = /var/scandata_exports/scandata_export_23.csv, Hash = sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\n","stream":"stdout","time":"2023-01-17T14:00:49.255823515Z"}
2023-01-17 15:00:49	
{"log":"2023-01-17T14:00:49Z [INFO] [/pkg/scan/export/filter_processor.go:65]: Selected 0 projects administered by user user@random.com \n","stream":"stdout","time":"2023-01-17T14:00:49.254222426Z"}
2023-01-17 15:00:49	
{"log":"2023-01-17T14:00:49Z [INFO] [/pkg/scan/export/filter_processor.go:234]: User user@random.com is not sys admin. Selecting projects with admin roles for export.\n","stream":"stdout","time":"2023-01-17T14:00:49.25244214Z"}
2023-01-17 15:00:49	
{"log":"2023-01-17T14:00:49Z [INFO] [/pkg/scan/export/filter_processor.go:53]: Retrieved user id :3 for user name : user@random.com\n","stream":"stdout","time":"2023-01-17T14:00:49.25238036Z"}
2023-01-17 15:00:49	
{"log":"2023-01-17T14:00:49Z [INFO] [/jobservice/job/impl/scandataexport/scan_data_export.go:179]: Request for export : map[CVEIds: JobName: Labels:[] Projects:[10] Repositories: Tags: UserID:3 UserName:user@random.com]\n","stream":"stdout","time":"2023-01-17T14:00:49.230163606Z"}
2023-01-17 15:00:49	
{"log":"2023-01-17T14:00:49Z [INFO] [/jobservice/job/impl/scandataexport/scan_data_export.go:173]: Created CSV export file /var/scandata_exports/scandata_export_23.csv\n","stream":"stdout","time":"2023-01-17T14:00:49.230130954Z"}
2023-01-17 15:00:49	
{"log":"2023-01-17T14:00:49Z [INFO] [/jobservice/job/impl/scandataexport/scan_data_export.go:78]: Scan data export job started in mode : export\n","stream":"stdout","time":"2023-01-17T14:00:49.210136741Z"}
2023-01-17 15:00:49	
{"log":"2023-01-17T14:00:49Z [INFO] [/pkg/config/rest/rest.go:47]: get configuration from url: https://harbor-core:443/api/v2.0/internalconfig\n","stream":"stdout","time":"2023-01-17T14:00:49.170683314Z"}
2023-01-17 15:00:49	
{"log":"2023-01-17T14:00:49Z [INFO] [/pkg/config/rest/rest.go:47]: get configuration from url: https://harbor-core:443/api/v2.0/internalconfig\n","stream":"stdout","time":"2023-01-17T14:00:49.157712531Z"}
2023-01-17 15:00:49	
{"log":"2023-01-17T14:00:49Z [INFO] [/jobservice/worker/cworker/c_worker.go:77]: Job incoming: {\"name\":\"SCAN_DATA_EXPORT\",\"id\":\"0cb9aa60bae2323ef4bbca5d\",\"t\":1673964049,\"args\":null}\n","stream":"stdout","time":"2023-01-17T14:00:49.156334965Z"}
```
This
`User user@random.com is not sys admin. Selecting projects with admin roles for export`
and this: 
`Selected 0 projects administered by user user@random.com`

Seem to indicate there is a problem with the verification of the role for this user, because this user **is** 'Project Admin'  for this project. 

When we did some more debugging we could get the CVE export function to work by doing the following:
In our current setup we assign Harbor roles to OIDC groups for each project, however if we assigned a Harbor role to the user directly, the CVE export function actually generated a report, see log files: 

```
2023-01-17 15:14:34	
{"log":"2023-01-17T14:14:34Z [INFO] [/jobservice/runner/redis.go:152]: Job 'SCAN_DATA_EXPORT:e5cfedc5e6548005c14e74bb' exit with success\n","stream":"stdout","time":"2023-01-17T14:14:34.170558575Z"}
2023-01-17 15:14:34	
{"log":"2023-01-17T14:14:34Z [INFO] [/jobservice/job/impl/scandataexport/scan_data_export.go:139]: Scan data export job completed\n","stream":"stdout","time":"2023-01-17T14:14:34.154705618Z"}
2023-01-17 15:14:34	
{"log":"2023-01-17T14:14:34Z [INFO] [/jobservice/job/impl/scandataexport/scan_data_export.go:133]: Export Job Id = 24. Created system artifact: 12 for report file /var/scandata_exports/scandata_export_24.csv to persistent storage: \u003cnil\u003e\n","stream":"stdout","time":"2023-01-17T14:14:34.146866305Z"}
2023-01-17 15:14:33	
{"log":"2023-01-17T14:14:33Z [INFO] [/pkg/config/rest/rest.go:47]: get configuration from url: https://harbor-core:443/api/v2.0/internalconfig\n","stream":"stdout","time":"2023-01-17T14:14:33.159602561Z"}
2023-01-17 15:14:33	
{"log":"2023-01-17T14:14:33Z [INFO] [/jobservice/job/impl/scandataexport/scan_data_export.go:111]: Export Job Id = 24. CSV file size: 814\n","stream":"stdout","time":"2023-01-17T14:14:33.156896997Z"}
2023-01-17 15:14:33	
{"log":"2023-01-17T14:14:33Z [INFO] [/jobservice/job/impl/scandataexport/scan_data_export.go:105]: Creating repository for CSV file with blob : scandata_export_24\n","stream":"stdout","time":"2023-01-17T14:14:33.156877854Z"}
2023-01-17 15:14:33	
{"log":"2023-01-17T14:14:33Z [INFO] [/jobservice/job/impl/scandataexport/scan_data_export.go:95]: Export Job Id = 24, FileName = /var/scandata_exports/scandata_export_24.csv, Hash = sha256:406646caabd1ec846b3011f851394223138fabf578e4c54e5d2ea7dca6920612\n","stream":"stdout","time":"2023-01-17T14:14:33.154570795Z"}
2023-01-17 15:14:33	
{"log":"2023-01-17T14:14:33Z [INFO] [/jobservice/job/impl/scandataexport/scan_data_export.go:263]: Export Group Id = 0, Job Id = 24, Page Number = 1, Page Size = 100000 Num Records = 2\n","stream":"stdout","time":"2023-01-17T14:14:33.145695742Z"}
2023-01-17 15:14:33	
{"log":"2023-01-17T14:14:33Z [INFO] [/pkg/scan/export/filter_processor.go:65]: Selected 1 projects administered by user user@random.com \n","stream":"stdout","time":"2023-01-17T14:14:33.128070209Z"}
2023-01-17 15:14:33	
{"log":"2023-01-17T14:14:33Z [INFO] [/pkg/scan/export/filter_processor.go:234]: User user@random.com is not sys admin. Selecting projects with admin roles for export.\n","stream":"stdout","time":"2023-01-17T14:14:33.125423744Z"}
2023-01-17 15:14:33	
{"log":"2023-01-17T14:14:33Z [INFO] [/pkg/scan/export/filter_processor.go:53]: Retrieved user id :3 for user name : user@random.com\n","stream":"stdout","time":"2023-01-17T14:14:33.125216538Z"}
2023-01-17 15:14:33	
{"log":"2023-01-17T14:14:33Z [INFO] [/jobservice/job/impl/scandataexport/scan_data_export.go:179]: Request for export : map[CVEIds: JobName: Labels:[] Projects:[10] Repositories: Tags: UserID:3 UserName:user@random.com]\n","stream":"stdout","time":"2023-01-17T14:14:33.101778451Z"}
2023-01-17 15:14:33	
{"log":"2023-01-17T14:14:33Z [INFO] [/jobservice/job/impl/scandataexport/scan_data_export.go:173]: Created CSV export file /var/scandata_exports/scandata_export_24.csv\n","stream":"stdout","time":"2023-01-17T14:14:33.101637254Z"}
2023-01-17 15:14:33	
{"log":"2023-01-17T14:14:33Z [INFO] [/jobservice/job/impl/scandataexport/scan_data_export.go:78]: Scan data export job started in mode : export\n","stream":"stdout","time":"2023-01-17T14:14:33.092637287Z"}
2023-01-17 15:14:33	
{"log":"2023-01-17T14:14:33Z [INFO] [/pkg/config/rest/rest.go:47]: get configuration from url: https://harbor-core:443/api/v2.0/internalconfig\n","stream":"stdout","time":"2023-01-17T14:14:33.058182951Z"}
2023-01-17 15:14:33	
{"log":"2023-01-17T14:14:33Z [INFO] [/pkg/config/rest/rest.go:47]: get configuration from url: https://harbor-core:443/api/v2.0/internalconfig\n","stream":"stdout","time":"2023-01-17T14:14:33.041614582Z"}
2023-01-17 15:14:33	
{"log":"2023-01-17T14:14:33Z [INFO] [/jobservice/worker/cworker/c_worker.go:77]: Job incoming: {\"name\":\"SCAN_DATA_EXPORT\",\"id\":\"e5cfedc5e6548005c14e74bb\",\"t\":1673964872,\"args\":null}\n","stream":"stdout","time":"2023-01-17T14:14:33.040151176Z"}
```
**Note** that it now reports 1 project instead of 0:
`Selected 1 projects administered by user user@random.com `

Our current setup with assigning roles to OIDC groups works for all other 'registry operations', however with the CVE export function there seems to be a problem.

**Expected behavior and actual behavior:**

_Expected behavior:_
User_A is member of project_A and is assigned the role 'Project Admin' for being member of the OIDC group group_A. 
User_A should be able to create/export a CVE report of repositories in project_A through the Harbor GUI.

_Actual behavior:_
User_A is member of project_A and is assigned the role 'Project Admin' for being member of the OIDC group group_A. 
User_A that generates CVE report for project_A (that contains CVEs) always gets the following error: 
`No vulnerabilities found or matched
`

**Versions:**
Please specify the versions of following systems.

- harbor version: [2.6.2]
- Running on EKS 1.23

--- a/src/controller/scandataexport/execution.go
+++ b/src/controller/scandataexport/execution.go
@@ -7,7 +7,6 @@ import (
 	"time"
 
 	"github.com/goharbor/harbor/src/jobservice/job"
-	"github.com/goharbor/harbor/src/jobservice/logger"
 	"github.com/goharbor/harbor/src/lib/errors"
 	"github.com/goharbor/harbor/src/lib/log"
 	"github.com/goharbor/harbor/src/lib/orm"
@@ -66,6 +65,7 @@ func (c *controller) ListExecutions(ctx context.Context, userName string) ([]*ex
 }
 
 func (c *controller) GetTask(ctx context.Context, executionID int64) (*task.Task, error) {
+	logger := log.GetLogger(ctx)
 	query := q2.New(q2.KeyWords{})
 
 	keywords := make(map[string]interface{})
@@ -90,6 +90,7 @@ func (c *controller) GetTask(ctx context.Context, executionID int64) (*task.Task
 }
 
 func (c *controller) GetExecution(ctx context.Context, executionID int64) (*export.Execution, error) {
+	logger := log.GetLogger(ctx)
 	exec, err := c.execMgr.Get(ctx, executionID)
 	if err != nil {
 		logger.Errorf("Error when fetching execution status for ExecutionId: %d error : %v", executionID, err)
@@ -103,6 +104,7 @@ func (c *controller) GetExecution(ctx context.Context, executionID int64) (*expo
 }
 
 func (c *controller) DeleteExecution(ctx context.Context, executionID int64) error {
+	logger := log.GetLogger(ctx)
 	err := c.execMgr.Delete(ctx, executionID)
 	if err != nil {
 		logger.Errorf("Error when deleting execution  for ExecutionId: %d, error : %v", executionID, err)
@@ -188,12 +190,17 @@ func (c *controller) convertToExportExecStatus(ctx context.Context, exec *task.E
 	if statusMessage, ok := exec.ExtraAttrs[export.StatusMessageAttribute]; ok {
 		execStatus.StatusMessage = statusMessage.(string)
 	}
-	artifactExists := c.isCsvArtifactPresent(ctx, exec.ID, execStatus.ExportDataDigest)
-	execStatus.FilePresent = artifactExists
+
+	if len(execStatus.ExportDataDigest) > 0 {
+		artifactExists := c.isCsvArtifactPresent(ctx, exec.ID, execStatus.ExportDataDigest)
+		execStatus.FilePresent = artifactExists
+	}
+
 	return execStatus
 }
 
 func (c *controller) isCsvArtifactPresent(ctx context.Context, execID int64, digest string) bool {
+	logger := log.GetLogger(ctx)
 	repositoryName := fmt.Sprintf("scandata_export_%v", execID)
 	exists, err := c.sysArtifactMgr.Exists(ctx, strings.ToLower(export.Vendor), repositoryName, digest)
 	if err != nil {

--- a/src/controller/scandataexport/execution_test.go
+++ b/src/controller/scandataexport/execution_test.go
@@ -97,6 +97,7 @@ func (suite *ScanDataExportExecutionTestSuite) TestGetExecution() {
 	attrs := make(map[string]interface{})
 	attrs[export.JobNameAttribute] = "test-job"
 	attrs[export.UserNameAttribute] = "test-user"
+	attrs[export.DigestKey] = "sha256:d04b98f48e8f8bcc15c6ae5ac050801cd6dcfd428fb5f9e65c4e16e7807340fa"
 	attrs["status_message"] = "test-message"
 	{
 		exec := task.Execution{

--- a/src/jobservice/job/impl/scandataexport/scan_data_export.go
+++ b/src/jobservice/job/impl/scandataexport/scan_data_export.go
@@ -184,14 +184,7 @@ func (sde *ScanDataExport) writeCsvFile(ctx job.Context, params job.Parameters, 
 			return err
 		}
 
-		// check if any projects are specified. If not then fetch all the projects
-		// of which the current user is a project admin.
-		projectIds, err := sde.filterProcessor.ProcessProjectFilter(systemContext, filterCriteria.UserName, filterCriteria.Projects)
-
-		if err != nil {
-			return err
-		}
-
+		projectIds := filterCriteria.Projects
 		if len(projectIds) == 0 {
 			return nil
 		}

--- a/src/jobservice/job/impl/scandataexport/scan_data_export_test.go
+++ b/src/jobservice/job/impl/scandataexport/scan_data_export_test.go
@@ -73,7 +73,6 @@ func (suite *ScanDataExportJobTestSuite) TestRun() {
 	data := suite.createDataRecords(3)
 	mock.OnAnything(suite.exportMgr, "Fetch").Return(data, nil).Once()
 	mock.OnAnything(suite.digestCalculator, "Calculate").Return(digest.Digest(MockDigest), nil)
-	mock.OnAnything(suite.filterProcessor, "ProcessProjectFilter").Return([]int64{1}, nil).Once()
 	mock.OnAnything(suite.filterProcessor, "ProcessRepositoryFilter").Return([]int64{1}, nil).Once()
 	mock.OnAnything(suite.filterProcessor, "ProcessTagFilter").Return([]*artifact.Artifact{{Artifact: artpkg.Artifact{ID: 1}}}, nil).Once()
 	mock.OnAnything(suite.filterProcessor, "ProcessLabelFilter").Return([]*artifact.Artifact{{Artifact: artpkg.Artifact{ID: 1}}}, nil).Once()
@@ -86,7 +85,9 @@ func (suite *ScanDataExportJobTestSuite) TestRun() {
 	params := job.Parameters{}
 	params[export.JobModeKey] = export.JobModeExport
 	params["JobId"] = JobId
-	params["Request"] = map[string]interface{}{}
+	params["Request"] = map[string]interface{}{
+		"projects": []int64{1},
+	}
 	ctx := &mockjobservice.MockJobContext{}
 
 	err := suite.job.Run(ctx, params)
@@ -138,7 +139,6 @@ func (suite *ScanDataExportJobTestSuite) TestRunAttributeUpdateError() {
 
 	data := suite.createDataRecords(3)
 	mock.OnAnything(suite.exportMgr, "Fetch").Return(data, nil).Once()
-	mock.OnAnything(suite.filterProcessor, "ProcessProjectFilter").Return([]int64{1}, nil).Once()
 	mock.OnAnything(suite.filterProcessor, "ProcessRepositoryFilter").Return([]int64{1}, nil).Once()
 	mock.OnAnything(suite.filterProcessor, "ProcessTagFilter").Return([]*artifact.Artifact{{Artifact: artpkg.Artifact{ID: 1}}}, nil).Once()
 	mock.OnAnything(suite.filterProcessor, "ProcessLabelFilter").Return([]*artifact.Artifact{{Artifact: artpkg.Artifact{ID: 1}}}, nil).Once()
@@ -152,7 +152,9 @@ func (suite *ScanDataExportJobTestSuite) TestRunAttributeUpdateError() {
 	params := job.Parameters{}
 	params[export.JobModeKey] = export.JobModeExport
 	params["JobId"] = JobId
-	params["Request"] = map[string]interface{}{}
+	params["Request"] = map[string]interface{}{
+		"projects": []int{1},
+	}
 	ctx := &mockjobservice.MockJobContext{}
 
 	err := suite.job.Run(ctx, params)
@@ -209,7 +211,6 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteria() {
 
 		repoCandidates := []int64{1}
 		artCandidates := []*artifact.Artifact{{Artifact: artpkg.Artifact{ID: 1, Digest: "digest1"}}}
-		mock.OnAnything(suite.filterProcessor, "ProcessProjectFilter").Return([]int64{1}, nil).Once()
 		mock.OnAnything(suite.filterProcessor, "ProcessRepositoryFilter").Return(repoCandidates, nil)
 		mock.OnAnything(suite.filterProcessor, "ProcessTagFilter").Return(artCandidates, nil)
 		mock.OnAnything(suite.filterProcessor, "ProcessLabelFilter").Return(artCandidates, nil)
@@ -274,7 +275,6 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteria() {
 
 		repoCandidate1 := &selector.Candidate{NamespaceID: 1}
 		repoCandidates := []*selector.Candidate{repoCandidate1}
-		mock.OnAnything(suite.filterProcessor, "ProcessProjectFilter").Return([]int64{1}, nil).Once()
 		mock.OnAnything(suite.filterProcessor, "ProcessRepositoryFilter").Return(repoCandidates, nil)
 		mock.OnAnything(suite.filterProcessor, "ProcessTagFilter").Return(repoCandidates, nil)
 
@@ -321,111 +321,6 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteria() {
 	}
 }
 
-func (suite *ScanDataExportJobTestSuite) TestRunWithCriteriaForProjectIdFilter() {
-	{
-		data := suite.createDataRecords(3)
-
-		mock.OnAnything(suite.exportMgr, "Fetch").Return(data, nil).Once()
-		mock.OnAnything(suite.exportMgr, "Fetch").Return(make([]export.Data, 0), nil).Once()
-		mock.OnAnything(suite.digestCalculator, "Calculate").Return(digest.Digest(MockDigest), nil)
-		mock.OnAnything(suite.exportMgr, "Fetch").Return(data, nil).Once()
-		mock.OnAnything(suite.exportMgr, "Fetch").Return(make([]export.Data, 0), nil).Once()
-		mock.OnAnything(suite.digestCalculator, "Calculate").Return(digest.Digest(MockDigest), nil)
-		execAttrs := make(map[string]interface{})
-		execAttrs[export.JobNameAttribute] = "test-job"
-		execAttrs[export.UserNameAttribute] = "test-user"
-		mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: int64(JobId), ExtraAttrs: execAttrs}, nil).Once()
-
-		mock.OnAnything(suite.filterProcessor, "ProcessProjectFilter").Return(nil, errors.New("test error")).Once()
-		mock.OnAnything(suite.filterProcessor, "ProcessRepositoryFilter").Return(nil, nil).Once()
-		mock.OnAnything(suite.filterProcessor, "ProcessTagFilter").Return(nil, nil).Once()
-
-		criteria := export.Request{
-			CVEIds:       "CVE-123",
-			Labels:       []int64{1},
-			Projects:     []int64{1},
-			Repositories: "test-repo",
-			Tags:         "test-tag",
-		}
-		criteriaMap := make(map[string]interface{})
-		bytes, _ := json.Marshal(criteria)
-		json.Unmarshal(bytes, &criteriaMap)
-		params := job.Parameters{}
-		params[export.JobModeKey] = export.JobModeExport
-		params["JobId"] = JobId
-		params["Request"] = criteriaMap
-
-		ctx := &mockjobservice.MockJobContext{}
-		ctx.On("SystemContext").Return(context.TODO()).Once()
-
-		err := suite.job.Run(ctx, params)
-		suite.Error(err)
-		sysArtifactRecordMatcher := testifymock.MatchedBy(func(sa *model.SystemArtifact) bool {
-			return sa.Repository == "scandata_export_100" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
-		})
-		suite.sysArtifactMgr.AssertNotCalled(suite.T(), "Create", mock.Anything, sysArtifactRecordMatcher, mock.Anything)
-		suite.execMgr.AssertNotCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, int64(JobId), mock.Anything)
-		_, err = os.Stat("/tmp/scandata_export_100.csv")
-
-		suite.exportMgr.AssertNotCalled(suite.T(), "Fetch", mock.Anything, mock.Anything)
-
-		suite.Truef(os.IsNotExist(err), "Expected CSV file to be deleted")
-	}
-
-	// empty list of projects
-	{
-		data := suite.createDataRecords(3)
-
-		mock.OnAnything(suite.exportMgr, "Fetch").Return(data, nil).Once()
-		mock.OnAnything(suite.exportMgr, "Fetch").Return(make([]export.Data, 0), nil).Once()
-		mock.OnAnything(suite.digestCalculator, "Calculate").Return(digest.Digest(MockDigest), nil)
-		mock.OnAnything(suite.exportMgr, "Fetch").Return(data, nil).Once()
-		mock.OnAnything(suite.exportMgr, "Fetch").Return(make([]export.Data, 0), nil).Once()
-		mock.OnAnything(suite.digestCalculator, "Calculate").Return(digest.Digest(MockDigest), nil)
-		execAttrs := make(map[string]interface{})
-		execAttrs[export.JobNameAttribute] = "test-job"
-		execAttrs[export.UserNameAttribute] = "test-user"
-		mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: int64(JobId), ExtraAttrs: execAttrs}, nil).Once()
-
-		mock.OnAnything(suite.filterProcessor, "ProcessProjectFilter").Return([]int64{}, nil).Once()
-		mock.OnAnything(suite.filterProcessor, "ProcessRepositoryFilter").Return([]int64{1}, nil).Once()
-		mock.OnAnything(suite.filterProcessor, "ProcessTagFilter").Return([]*artifact.Artifact{{Artifact: artpkg.Artifact{ID: 1}}}, nil).Once()
-
-		criteria := export.Request{
-			CVEIds:       "CVE-123",
-			Labels:       []int64{1},
-			Projects:     []int64{1},
-			Repositories: "test-repo",
-			Tags:         "test-tag",
-		}
-		criteriaMap := make(map[string]interface{})
-		bytes, _ := json.Marshal(criteria)
-		json.Unmarshal(bytes, &criteriaMap)
-		params := job.Parameters{}
-		params[export.JobModeKey] = export.JobModeExport
-		params["JobId"] = JobId
-		params["Request"] = criteriaMap
-
-		ctx := &mockjobservice.MockJobContext{}
-		ctx.On("SystemContext").Return(context.TODO()).Once()
-
-		err := suite.job.Run(ctx, params)
-		suite.NoError(err)
-		sysArtifactRecordMatcher := testifymock.MatchedBy(func(sa *model.SystemArtifact) bool {
-			return sa.Repository == "scandata_export_100" && sa.Vendor == strings.ToLower(export.Vendor) && sa.Digest == MockDigest
-		})
-		suite.sysArtifactMgr.AssertNotCalled(suite.T(), "Create", mock.Anything, sysArtifactRecordMatcher, mock.Anything)
-
-		suite.execMgr.AssertCalled(suite.T(), "UpdateExtraAttrs", mock.Anything, int64(JobId), mock.Anything)
-		_, err = os.Stat("/tmp/scandata_export_100.csv")
-
-		suite.exportMgr.AssertNotCalled(suite.T(), "Fetch", mock.Anything, mock.Anything)
-
-		suite.Truef(os.IsNotExist(err), "Expected CSV file to be deleted")
-	}
-
-}
-
 func (suite *ScanDataExportJobTestSuite) TestRunWithCriteriaForRepositoryIdFilter() {
 	{
 		data := suite.createDataRecords(3)
@@ -441,7 +336,6 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteriaForRepositoryIdFilte
 		execAttrs[export.UserNameAttribute] = "test-user"
 		mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: int64(JobId), ExtraAttrs: execAttrs}, nil).Once()
 
-		mock.OnAnything(suite.filterProcessor, "ProcessProjectFilter").Return([]int64{1}, errors.New("test error")).Once()
 		mock.OnAnything(suite.filterProcessor, "ProcessRepositoryFilter").Return([]int64{1}, errors.New("test error")).Once()
 		mock.OnAnything(suite.filterProcessor, "ProcessTagFilter").Return([]*artifact.Artifact{{Artifact: artpkg.Artifact{ID: 1}}}, nil).Once()
 
@@ -492,7 +386,6 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteriaForRepositoryIdFilte
 		execAttrs[export.UserNameAttribute] = "test-user"
 		mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: int64(JobId), ExtraAttrs: execAttrs}, nil).Once()
 
-		mock.OnAnything(suite.filterProcessor, "ProcessProjectFilter").Return([]int64{}, nil).Once()
 		mock.OnAnything(suite.filterProcessor, "ProcessRepositoryFilter").Return([]int64{}, nil).Once()
 		mock.OnAnything(suite.filterProcessor, "ProcessTagFilter").Return([]*artifact.Artifact{}, nil).Once()
 
@@ -545,7 +438,6 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteriaForRepositoryIdWithT
 		execAttrs[export.UserNameAttribute] = "test-user"
 		mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: int64(JobId), ExtraAttrs: execAttrs}, nil).Once()
 
-		mock.OnAnything(suite.filterProcessor, "ProcessProjectFilter").Return([]int64{1}, errors.New("test error")).Once()
 		mock.OnAnything(suite.filterProcessor, "ProcessRepositoryFilter").Return([]int64{1}, nil).Once()
 		mock.OnAnything(suite.filterProcessor, "ProcessTagFilter").Return(nil, errors.New("test error")).Once()
 
@@ -596,7 +488,6 @@ func (suite *ScanDataExportJobTestSuite) TestRunWithCriteriaForRepositoryIdWithT
 		execAttrs[export.UserNameAttribute] = "test-user"
 		mock.OnAnything(suite.execMgr, "Get").Return(&task.Execution{ID: int64(JobId), ExtraAttrs: execAttrs}, nil).Once()
 
-		mock.OnAnything(suite.filterProcessor, "ProcessProjectFilter").Return([]int64{}, nil).Once()
 		mock.OnAnything(suite.filterProcessor, "ProcessRepositoryFilter").Return([]int64{}, nil).Once()
 		mock.OnAnything(suite.filterProcessor, "ProcessTagFilter").Return(nil, nil).Once()
 

--- a/src/pkg/scan/export/filter_processor.go
+++ b/src/pkg/scan/export/filter_processor.go
@@ -3,24 +3,19 @@ package export
 import (
 	"context"
 
-	commonmodels "github.com/goharbor/harbor/src/common/models"
-	"github.com/goharbor/harbor/src/common/security/local"
 	"github.com/goharbor/harbor/src/common/utils"
 	"github.com/goharbor/harbor/src/controller/artifact"
-	"github.com/goharbor/harbor/src/jobservice/logger"
 	"github.com/goharbor/harbor/src/lib/q"
 	"github.com/goharbor/harbor/src/lib/selector"
 	"github.com/goharbor/harbor/src/lib/selector/selectors/doublestar"
 	"github.com/goharbor/harbor/src/pkg"
 	artpkg "github.com/goharbor/harbor/src/pkg/artifact"
 	"github.com/goharbor/harbor/src/pkg/project"
-	"github.com/goharbor/harbor/src/pkg/project/models"
 	"github.com/goharbor/harbor/src/pkg/repository"
 	"github.com/goharbor/harbor/src/pkg/user"
 )
 
 type FilterProcessor interface {
-	ProcessProjectFilter(ctx context.Context, userName string, projectsToFilter []int64) ([]int64, error)
 	ProcessRepositoryFilter(ctx context.Context, filter string, projectIds []int64) ([]int64, error)
 	ProcessTagFilter(ctx context.Context, filter string, repositoryIds []int64) ([]*artifact.Artifact, error)
 	ProcessLabelFilter(ctx context.Context, labelIDs []int64, arts []*artifact.Artifact) ([]*artifact.Artifact, error)
@@ -41,50 +36,6 @@ func NewFilterProcessor() FilterProcessor {
 		usrMgr:     user.Mgr,
 		projectMgr: pkg.ProjectMgr,
 	}
-}
-
-func (dfp *DefaultFilterProcessor) ProcessProjectFilter(ctx context.Context, userName string, projectIdsToFilter []int64) ([]int64, error) {
-	// get the user id of the current user
-
-	usr, err := dfp.usrMgr.GetByName(ctx, userName)
-	if err != nil {
-		return nil, err
-	}
-	logger.Infof("Retrieved user id :%d for user name : %s", usr.UserID, userName)
-
-	if err != nil {
-		return nil, err
-	}
-
-	query := dfp.getProjectQueryFilter(usr)
-	projects, err := dfp.projectMgr.List(ctx, query)
-
-	if err != nil {
-		return nil, err
-	}
-	logger.Infof("Selected %d projects administered by user %s ", len(projects), userName)
-	projectIds := make([]int64, 0)
-	for _, proj := range projects {
-		projectIds = append(projectIds, proj.ProjectID)
-	}
-
-	// check if the project ids specified in the filter are present in the list
-	// of projects of which the current user is a project admin
-	if len(projectIdsToFilter) == 0 {
-		return projectIds, nil
-	}
-	m := make(map[int64]bool)
-	for _, projectID := range projectIds {
-		m[projectID] = true
-	}
-	filtered := make([]int64, 0)
-
-	for _, filteredProjID := range projectIdsToFilter {
-		if m[filteredProjID] {
-			filtered = append(filtered, filteredProjID)
-		}
-	}
-	return filtered, nil
 }
 
 func (dfp *DefaultFilterProcessor) ProcessRepositoryFilter(ctx context.Context, filter string, projectIds []int64) ([]int64, error) {
@@ -222,15 +173,4 @@ func (dfp *DefaultFilterProcessor) ProcessLabelFilter(ctx context.Context, label
 	}
 
 	return filteredArts, nil
-}
-
-func (dfp *DefaultFilterProcessor) getProjectQueryFilter(user *commonmodels.User) *q.Query {
-	secContext := local.NewSecurityContext(user)
-	if secContext.IsSysAdmin() {
-		logger.Infof("User %v is sys admin. Selecting all projects for export.", user.Username)
-		return q.New(q.KeyWords{})
-	}
-
-	logger.Infof("User %v is not sys admin. Selecting projects with admin roles for export.", user.Username)
-	return q.New(q.KeyWords{"member": &models.MemberQuery{UserID: user.UserID, GroupIDs: user.GroupIDs}})
 }

--- a/src/pkg/scan/export/filter_processor_test.go
+++ b/src/pkg/scan/export/filter_processor_test.go
@@ -3,21 +3,15 @@ package export
 import (
 	"context"
 	"errors"
-	"reflect"
 	"testing"
 	"time"
 
-	testifymock "github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/suite"
 
-	commonmodels "github.com/goharbor/harbor/src/common/models"
 	"github.com/goharbor/harbor/src/controller/artifact"
-	project3 "github.com/goharbor/harbor/src/controller/project"
 	"github.com/goharbor/harbor/src/controller/tag"
-	"github.com/goharbor/harbor/src/lib/q"
 	artpkg "github.com/goharbor/harbor/src/pkg/artifact"
 	labelmodel "github.com/goharbor/harbor/src/pkg/label/model"
-	"github.com/goharbor/harbor/src/pkg/project/models"
 	"github.com/goharbor/harbor/src/pkg/repository/model"
 	tagmodel "github.com/goharbor/harbor/src/pkg/tag/model/tag"
 	artifactctl "github.com/goharbor/harbor/src/testing/controller/artifact"
@@ -51,70 +45,6 @@ func (suite *FilterProcessorTestSuite) SetupTest() {
 		usrMgr:     suite.usrMgr,
 		projectMgr: suite.projectMgr,
 	}
-}
-
-func (suite *FilterProcessorTestSuite) TestProcessProjectFilter() {
-	project1 := &models.Project{ProjectID: 1}
-
-	project2 := &models.Project{ProjectID: 2}
-
-	// no filtered projects returns all projects
-	{
-		suite.usrMgr.On("GetByName", mock.Anything, "test-user").Return(&commonmodels.User{UserID: 1}, nil).Once()
-		suite.projectMgr.On("List", mock.Anything, mock.Anything).Return([]*models.Project{project1, project2}, nil).Once()
-		projectIds, err := suite.filterProcessor.ProcessProjectFilter(context.TODO(), "test-user", []int64{})
-		suite.Equal(2, len(projectIds))
-		suite.NoError(err)
-	}
-
-	// filtered project
-	{
-		suite.usrMgr.On("GetByName", mock.Anything, "test-user").Return(&commonmodels.User{UserID: 1}, nil).Once()
-		suite.projectMgr.On("List", mock.Anything, mock.Anything).Return([]*models.Project{project1, project2}, nil).Once()
-		projectIds, err := suite.filterProcessor.ProcessProjectFilter(context.TODO(), "test-user", []int64{1})
-		suite.Equal(1, len(projectIds))
-		suite.Equal(int64(1), projectIds[0])
-		suite.NoError(err)
-	}
-
-	// filtered project with group ids
-	{
-		groupIDs := []int{4, 5}
-		suite.usrMgr.On("GetByName", mock.Anything, "test-user").Return(&commonmodels.User{UserID: 1, GroupIDs: groupIDs}, nil).Once()
-		suite.projectMgr.On("List", mock.Anything, mock.Anything).Return([]*models.Project{project1, project2}, nil).Once()
-		projectIds, err := suite.filterProcessor.ProcessProjectFilter(context.TODO(), "test-user", []int64{1})
-		suite.Equal(1, len(projectIds))
-		suite.Equal(int64(1), projectIds[0])
-		suite.NoError(err)
-		memberQueryMatcher := testifymock.MatchedBy(func(query *q.Query) bool {
-			memberQuery := query.Keywords["member"].(*project3.MemberQuery)
-			return len(memberQuery.GroupIDs) == 2 && reflect.DeepEqual(memberQuery.GroupIDs, groupIDs) && memberQuery.Role == 0
-		})
-		suite.projectMgr.AssertCalled(suite.T(), "List", mock.Anything, memberQueryMatcher)
-	}
-
-	// project listing for admin user
-	{
-		suite.usrMgr.On("GetByName", mock.Anything, "test-user").Return(&commonmodels.User{UserID: 1, SysAdminFlag: true}, nil).Once()
-		suite.projectMgr.On("List", mock.Anything, mock.Anything).Return([]*models.Project{project1, project2}, nil).Once()
-		_, err := suite.filterProcessor.ProcessProjectFilter(context.TODO(), "test-user", []int64{1})
-		suite.NoError(err)
-		queryArgumentMatcher := testifymock.MatchedBy(func(query *q.Query) bool {
-			return len(query.Keywords) == 0
-		})
-		suite.projectMgr.AssertCalled(suite.T(), "List", mock.Anything, queryArgumentMatcher)
-	}
-
-	// project listing returns an error
-	// filtered project
-	{
-		suite.usrMgr.On("GetByName", mock.Anything, "test-user").Return(&commonmodels.User{UserID: 1}, nil).Once()
-		suite.projectMgr.On("List", mock.Anything, mock.Anything).Return(nil, errors.New("test-error")).Once()
-		projectIds, err := suite.filterProcessor.ProcessProjectFilter(context.TODO(), "test-user", []int64{1})
-		suite.Error(err)
-		suite.Nil(projectIds)
-	}
-
 }
 
 func (suite *FilterProcessorTestSuite) TestProcessRepositoryFilter() {

--- a/src/server/v2.0/handler/scanexport.go
+++ b/src/server/v2.0/handler/scanexport.go
@@ -54,8 +54,10 @@ func (se *scanDataExportAPI) ExportScanData(ctx context.Context, params operatio
 		return se.SendError(ctx, err)
 	}
 
-	if err := se.RequireProjectAccess(ctx, params.Criteria.Projects[0], rbac.ActionCreate, rbac.ResourceExportCVE); err != nil {
-		return se.SendError(ctx, err)
+	for _, pid := range params.Criteria.Projects {
+		if err := se.RequireProjectAccess(ctx, pid, rbac.ActionCreate, rbac.ResourceExportCVE); err != nil {
+			return se.SendError(ctx, err)
+		}
 	}
 
 	scanDataExportJob := new(models.ScanDataExportJob)

--- a/src/testing/pkg/scan/export/filter_processor.go
+++ b/src/testing/pkg/scan/export/filter_processor.go
@@ -38,29 +38,6 @@ func (_m *FilterProcessor) ProcessLabelFilter(ctx context.Context, labelIDs []in
 	return r0, r1
 }
 
-// ProcessProjectFilter provides a mock function with given fields: ctx, userName, projectsToFilter
-func (_m *FilterProcessor) ProcessProjectFilter(ctx context.Context, userName string, projectsToFilter []int64) ([]int64, error) {
-	ret := _m.Called(ctx, userName, projectsToFilter)
-
-	var r0 []int64
-	if rf, ok := ret.Get(0).(func(context.Context, string, []int64) []int64); ok {
-		r0 = rf(ctx, userName, projectsToFilter)
-	} else {
-		if ret.Get(0) != nil {
-			r0 = ret.Get(0).([]int64)
-		}
-	}
-
-	var r1 error
-	if rf, ok := ret.Get(1).(func(context.Context, string, []int64) error); ok {
-		r1 = rf(ctx, userName, projectsToFilter)
-	} else {
-		r1 = ret.Error(1)
-	}
-
-	return r0, r1
-}
-
 // ProcessRepositoryFilter provides a mock function with given fields: ctx, filter, projectIds
 func (_m *FilterProcessor) ProcessRepositoryFilter(ctx context.Context, filter string, projectIds []int64) ([]int64, error) {
 	ret := _m.Called(ctx, filter, projectIds)

